protocol/river-control-unstable-v1.xml

```diff
@@ -37,49 +29,421 @@
+      </description>
+      <arg name="seat" type="object" interface="wl_seat"/>
+      <arg name="axis" type="uint" enum="axis" summary="axis to resize along"/>
+      <arg name="delta" type="int" summary="amount/direction along the axis"/>
+    </request>
+
+    <enum name="edge">
+      <entry name="top" value="0"/>
+      <entry name="bottom" value="1"/>
+      <entry name="left" value="2"/>
+      <entry name="right" value="3"/>
+    </enum>
+
+    <request name="snap">
+      <description summary="snap the focused view to a screen edge">
+        The view is set to floating if not already floating.
+      </description>
+      <arg name="seat" type="object" interface="wl_seat"/>
+      <arg name="edge" type="uint" enum="edge" summary="edge to snap to"/>
+    </request>
+
+    <request name="layout">
+      <description summary="set the layout command">
+        Provide a command which river will use for generating the layout
+        of non-floating windows on the currently focused output. See
+        *river-layouts*(7) for details on the expected formatting of the
+        output of layout commands. Alternatively, "full" can be given
+        instead of a command to cause river to use its single internal layout,
+        in which windows span the entire width and height of the output.
+      </description>
+      <arg name="seat" type="object" interface="wl_seat"/>
+      <arg name="command" type="string" summary="the layout command"/>
+    </request>
+
+    <enum name="attach_mode">
+      <entry name="top" value="0"/>
+      <entry name="bottom" value="1"/>
+    </enum>
+
+    <request name="attach_mode">
+      <description summary="set the attach mode for the focused output">
+        Set where new views should attach to the view stack for the currently
+        focused output.
       </description>
       <arg name="seat" type="object" interface="wl_seat"/>
-      <arg name="callback" type="new_id" interface="zriver_command_callback_v1"
-        summary="callback object"/>
+      <arg name="mode" type="uint" enum="attach_mode"/>
+    </request>
+
+    <request name="csd_filter_add">
+      <description summary="add an app_id to the csd filter list">
+        Views with this app-id are allowed to use client side decorations
+        instead of the default server side decorations.
+      </description>
+      <arg name="app_id" type="string" summary="the app_id to filter"/>
+    </request>
+
+    <request name="float_filter_add">
+      <description summary="add an app_id to the float filter list">
+        Views with this app-id will start floating by default
+      </description>
+      <arg name="app_id" type="string" summary="the app_id to filter"/>
+    </request>
+
+    <request name="background_color">
+      <description summary="set the background color"/>
+      <arg name="red" type="fixed"/>
+      <arg name="green" type="fixed"/>
+      <arg name="blue" type="fixed"/>
+      <arg name="alpha" type="fixed"/>
```

Comment:
I don't think alpha makes much sense for the background colour.

Reply:
It's trivial to support and has an effect in a nested context so why not continue supporting it (we already do).